### PR TITLE
Tests get survey items

### LINKED ISSUE #15: API: getSurveyItems
### when a user asks for information about products (request `operation` is `'products'`)
  1. need to be able to request products by article-type ('shoe', 'accessory', 'top', 'bottom', 'dress' and 'hat')
  2. modify response to include a `style` (sporty, casual, edgy etc) property and `rating` in that style if applicable.
  3. need to populate at least 1 seed of each type of article for each style for all genders
  - possible backend entry-file: `/code/api/src/modules/product/resolvers`
  - frontend action location: `/code/web/src/modules/product/api/actions` 

- [x] Add a seed for each type
- [x] Create an endpoint that sends only the products to be used in the survey
- [x] Allow the FE to request survey items by type (top, bottom, accessory, etc.) Keep it simple for now
- [x] Add happy/sad path testing

### PR DESCRIPTION
**Pull request by D-Lessenden**
### This PRs additions:

Create query tests for the 4 types of getSurveyItems. Top, bottom, accessory, and full.

It also restructured a user query test, ```returns a user based on id```, to avoid an arbitrary failing test. Before the restructure the test was failing because the ``createdAt``` attribute for the user was different for each person on the team. 

 
### Reviewer starting point: 

If you have not seeded your database since the survey styles were added run the following in your terminal:

```NODE_ENV=test npm run db:seed; ```

### CODE REVIEW

File: code/api/src/modules/tests/user/query.test.js [hunk 1]
```diff
@@ -21,16 +21,10 @@ describe('users query', () => {
...
-          name: 'The Admin',
-          role: 'ADMIN',
-          createdAt: '1607446829755',
-        }
-      }
-    });
+      expect(response.body.data.user.email).toBe('admin@crate.com')
+      expect(response.body.data.user.name).toBe('The Admin')
+      expect(response.body.data.user.role).toBe('ADMIN')
+      expect(response.body.data.user).toHaveProperty('createdAt')
```
Comment (hashmaster3k) [inline code/api/src/modules/tests/user/query.test.js:27]: Thanks for changing that! Been meaning to get to it.

**Review by hashmaster3k — CHANGES_REQUESTED** [review]
Thanks for doing this Daniel! Looks good! I'd like to see a test for the values of the properties to make sure they have something being returned and not null. Looks like jest provides an expectation for a property to be truthy. That is what I am using.

File: code/api/src/modules/tests/survey/query.test.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      .expect(200)
+      expect(response.body.data.getSurveyItems.length).toBe(18)
+      expect(response.body.data.getSurveyItems[0]).toHaveProperty('id')
+      expect(response.body.data.getSurveyItems[0]).toHaveProperty('type')
+      expect(response.body.data.getSurveyItems[0]).toHaveProperty('score')
+      expect(response.body.data.getSurveyItems[0]).toHaveProperty('image')
+      expect(response.body.data.getSurveyItems[0].id).toBe(2)
+      expect(response.body.data.getSurveyItems[0].type).toBe('accessory')
+      expect(response.body.data.getSurveyItems[0].image).toBe("public/images/survey/casual/mens_shoes_casual.jpg")
+    done();
```
Comment (lkriffell) [inline code/api/src/modules/tests/survey/query.test.js:78]: I wonder if we could use a `forEach` loop  over `response.body.data.getSurveyItems` to write shorter tests that actually test that no item has a null attribute

Comment (D-Lessenden) [reply code/api/src/modules/tests/survey/query.test.js:78]: That is a great suggestion. I will implement that tomorrow. Thank you for the suggestion.

File: code/api/src/modules/tests/survey/query.test.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      results.forEach(result => expect(result.type).toBe('top'))
+      results.forEach(result => expect(result.type).not.toBe(null))
+
+      results.forEach(result => expect(result).toHaveProperty('score'))
+      results.forEach(result => expect(result.score).not.toBe(null))
+
+      results.forEach(result => expect(result).toHaveProperty('image'))
+      results.forEach(result => expect(result.image).not.toBe(null))
+
+      expect(response.body.data.getSurveyItems.length).toBe(8)
```
Comment (lkriffell) [inline code/api/src/modules/tests/survey/query.test.js:30]: Nice job! Looks great

**Review by lkriffell — APPROVED** [review]

**Review by hashmaster3k — APPROVED** [review]

**Pull request merged**